apps/anomalist/anomalist_api.py

```diff
@@ -0,0 +1,406 @@
+import tempfile
+from pathlib import Path
+from typing import Dict, List, Literal, Optional, Tuple, get_args
+
+import numpy as np
+import pandas as pd
+import structlog
+from owid.catalog import find
+from owid.datautils.dataframes import map_series
+from sqlalchemy.engine import Engine
+from sqlalchemy.orm import Session
+
+import etl.grapher_io as io
+from apps.anomalist.detectors import AnomalyTimeChange, AnomalyUpgradeChange, AnomalyUpgradeMissing
+from apps.wizard.utils.paths import WIZARD_ANOMALIES_RELATIVE
+from etl import grapher_model as gm
+from etl.config import OWID_ENV
+from etl.db import get_engine, read_sql
+from etl.files import create_folder, upload_file_to_server
+from etl.grapher_io import variable_data_df_from_s3
+
+log = structlog.get_logger()
+
+# Name of index columns for dataframe.
+INDEX_COLUMNS = ["entity_id", "year"]
+
+# Define anomaly types.
+ANOMALY_TYPE = Literal["upgrade_change", "time_change", "upgrade_missing"]
+
+# Define mapping of available anomaly types to anomaly detectors.
+ANOMALY_DETECTORS = {
+    "time_change": AnomalyTimeChange,
+    "upgrade_change": AnomalyUpgradeChange,
+    "upgrade_missing": AnomalyUpgradeMissing,
+}
+
+########################################################################################################################
+
+# AGGREGATE ANOMALIES:
+
+
+def load_latest_population():
+    # NOTE: The "channels" parameter of the find function is not working well.
+    candidates = find("population", channels=("grapher",), dataset="population", namespace="demography").sort_values(
```

Comment:
We could also load it from the local catalog with
```
population = Dataset(DATA_DIR / "grapher/owid/latest/key_indicators")["population"]
```
Might be better than `find`.